index.bs

```diff
@@ -706,30 +706,15 @@ occurred on the [=remote end=].
    are run when multiple events are enabled at once, with lower integers
    indicating steps that run earlier.
 
-A [=BiDi session=] has a <dfn export for=event>global event set</dfn>
-which is a [=/set=] containing the event names for events that are enabled for all
-navigables. This initially contains the [=event name=] for events that
-are <dfn export for=event>in the default event set</dfn>.
-
-A [=BiDi session=] has a <dfn export for=event>navigable event map</dfn>,
-which is a [=/map=] with [=/top-level traversable=] keys and values
-that are a [=/set=] of [=event name=]s for events that are enabled in the given
-navigable.
-
-<div algorithm>
-
-To obtain a list of <dfn>event enabled navigables</dfn> given
-|session| and |event name|:
-
-1. Let |navigables| be an empty [=/set=].
+A [=BiDi session=] has a <dfn export for=event>subscriptions</dfn>
+which is a [=/list=] of [=subscription=].
 
-1. For each |navigable| → |events| of |session|'s [=navigable event map=]:
+A <dfn for=event>subscription</dfn> is a [=/struct=] consisting of a
+subscription id (a string), event names (a [=/list=] of event names) and
+contexts (a [=/list=] of context ids).
```

Comment:
I don't think you have to include subscribing to events for an entire user context in this patch, but it might be useful to think about how that would fit into this data structure. Infra doesn't naturally define sum types (and I guess there's a question about whether we would allow a single subscription request to include both a list of top-level traversable ids — i.e. context ids — and a list of user context ids, or require one-or-the-other), but maybe we should have something like a "subscription type" field that can either be "global" or "context" or "user context" and a "subscription data" (or something) field that's a list of contexts or a list of user contexts, or always null for global subscriptions.

Reply:
I was thinking to add user contexts as another set on the subscription with empty sets indicating "all" for any fields here. 

So the subscription is type "global" iff empty(user contexts) and empty(contexts), "context" iff !empty(contexts) && empty(user contexts), "user context" iff  empty(contexts) && !empty(contexts). The structs also restrict that the attributes have to be of same type which I think means we cannot have a null. Subscriptions that have more than 1 non-empty set would be invalid (but not enforced by the types) for now. Alternatively, we can make a map out of the subscription.

Reply:
That approach seems fine to me.